MsApplicationPkg/SecureBootRecovery/SecureBootRecovery.c

```diff
@@ -11,15 +11,64 @@
 #include <Guid/ImageAuthentication.h>
 #include <Guid/VariableFormat.h>
 #include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
 
 #include "RecoveryPayload.h"
 
+//
 // 10 seconds in microseconds
+//
 #define STALL_10_SECONDS  10000000
 
+//
+// The size of an EFI_STATUS in bytes
+//
 #define STATUS_SIZE         (sizeof(EFI_STATUS) * 2)
 #define STATUS_STRING_SIZE  (STATUS_SIZE + sizeof(L"\0"))
 
+//
+// The certificate we are looking for
+//
+#define CERT_ORGANIZATION         "Microsoft Corporation"
+#define CERT_ORGANIZATION_OFFSET  0xF6
+#define CERT_COMMON_NAME          "Windows UEFI CA 2023"
+#define CERT_COMMON_NAME_OFFSET   0x116
+#define CERT_SIGNATURE            { 0x9f, 0xc9, 0xb6, 0xff, 0x6e, 0xe1, 0x9c, 0x3b, 0x55, 0xf6, 0xfe, 0x8b, 0x39, 0xdd, 0x61, 0x04 }
+#define CERT_SIGNATURE_OFFSET     0x3AE
+
+//
+// The system table pointers
+//
+EFI_SYSTEM_TABLE      *mST = NULL;
+EFI_RUNTIME_SERVICES  *mRT = NULL;
+EFI_BOOT_SERVICES     *mBS = NULL;

```

Comment:
Why not just use UefiBootServicesTableLib and UefiRuntimeServicesTableLib? They just maintain the pointers for you.

I understand dependencies are trying to be reduced but you're already dependent on much more complex logic elsewhere (like the functions backing those pointers) to the point the code redundancy vs complexity tradeoff I think favors just using the MdePkg library instances.

Reply:
Since this is going to be consumed by the Boot team and built with windows tooling instead of UEFI ultimately, I'm avoiding the additional libraries

Reply:
In that case, I suggest leaving a note saying why this is done since it is against typical edk2 development patterns.